src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/inlinestats/validate.test.ts

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { mockContext } from '../../../__tests__/context_fixtures';
+import { expectErrors } from '../../../__tests__/validation';
+import { validate } from '../stats/validate';
+
+const inlinestatsExpectErrors = (
+  query: string,
+  expectedErrors: string[],
+  context = mockContext
+) => {
+  return expectErrors(query, expectedErrors, context, 'inlinestats', validate);
+};
+
+describe('INLINESTATS Validation', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('INLINESTATS <aggregates> [ BY <grouping> ]', () => {
+    const newUserDefinedColumns = new Map(mockContext.userDefinedColumns);
+    newUserDefinedColumns.set('doubleField * 3.281', [
+      {
+        name: 'doubleField * 3.281',
+        type: 'double',
+        location: { min: 0, max: 10 },
+      },
+    ]);
+    newUserDefinedColumns.set('avg_doubleField', [
+      {
+        name: 'avg_doubleField',
+        type: 'double',
+        location: { min: 0, max: 10 },
+      },
+    ]);
+    const context = {
+      ...mockContext,
+      userDefinedColumns: newUserDefinedColumns,
+    };
+    test('requires aggregate before BY; other valid usages pass', () => {
+      inlinestatsExpectErrors('from a_index | INLINESTATS by textField', [
+        'Expected an aggregate function or group but got [textField] of type [FieldAttribute]',
```

Comment:
I think this message is ok for this case

Reply:
It is ok but it doesnt work for all cases. For example here you take this ugly message

<img width="1408" height="175" alt="image" src="https://github.com/user-attachments/assets/4f07a0be-58aa-410d-ab4b-ebce8e8644eb" />

But this comes from ES, not our responsibility to fix

Reply:
Looking closer, this is actually a false positive. We trigger an error here because of textField.
The reality is that we are not catching grammar errors.

We can  add small logic here src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/validation/commands.ts to handle this case with a proper and controlled message.

or...
 we skip these test cases and leave to ES this responsability, as you said

Reply:
Why because of textField?